src/relayserver/RelayServer.ts

```diff
@@ -432,20 +444,26 @@ export class RelayServer extends EventEmitter {
   }
 
   async getManagerBalance (): Promise<BN> {
-    return toBN(await this.contractInteractor.getBalance(this.managerAddress))
+    return toBN(await this.contractInteractor.getBalance(this.managerAddress, 'pending'))
   }
 
   async getWorkerBalance (workerIndex: number): Promise<BN> {
-    return toBN(await this.contractInteractor.getBalance(this.workerAddress))
+    return toBN(await this.contractInteractor.getBalance(this.workerAddress, 'pending'))
   }
 
   async _shouldRegisterAgain (currentBlock: number, hubEventsSinceLastScan: EventData[]): Promise<boolean> {
+    const isPendingActivityTransaction =
+      (await this.txStoreManager.isActionPending(ServerAction.RELAY_CALL)) ||
```

Comment:
relaycall should not interfere with other operation - its for a worker, not mgr.

Reply:
We had previously defined an option for 'relay call' event to be a relay discovery event.
Until we make a formal decision to change that (with Jira issue) I think we should keep it as was originally designed.